assets/js/blocks/collection-filters/inner-blocks/stock-filter/block.json

```diff
@@ -0,0 +1,46 @@
+{
+	"name": "woocommerce/collection-stock-filter",
+	"version": "1.0.0",
+	"title": "Stock Filter",
+	"description": "Enable customers to filter the product collection by stock status.",
+	"category": "woocommerce",
+	"keywords": [ "WooCommerce", "filter", "stock" ],
+	"supports": {
+		"interactivity": true,
+		"html": false,
+		"multiple": false
+	},
+	"attributes": {
+		"className": {
+			"type": "string",
+			"default": ""
+		},
+		"showCounts": {
+			"type": "boolean",
+			"default": false
+		},
+		"displayStyle": {
+			"type": "string",
+			"default": "list"
+		},
+		"selectType": {
+			"type": "string",
+			"default": "multiple"
+		},
```

Comment:
Yeah, so I think we need to implement this but can be done as follow up. I wanted to get the dropdown in because i know its needed for other things, then I can implement this feature along with simplifying the dropdown markup. Because the current iteration is just a port of the original one, I think it will get too complex to port the existing one, rather I would simplify it and add features like this as a follow up.

Reply:
I agree, let's work on it in a follow up PR.